### DISCUSSION TITLE
wldash lighweight launcher + dashboard /status bar / panel combo written in Rust 🦀

### DISCUSSION THREAD

**Discussion by HEX0101** [YaLTeR/niri#473] [General]
Hello,
I found out about wldash.
https://git.sr.ht/~kennylevinsen/wldash

![screenshot](https://github.com/YaLTeR/niri/assets/89577795/c79769ee-51cb-4bdf-a4ef-9b217a9cb57d)

It's a launcher plus dashboard combined and it's written in Rust 🦀

Unfortunately it hasn't been maintained for over a year but still works quite well. It's in archlinux repos so you can try it out without compiling.

I noticed waybar widgets usually poll in certain time intervals. I'm not sure if every DE panel does this but it seems very CPU inefficient to poll system volume every 100 miliseconds and display it correctly. Or system temperature, clock and so on. If you have a lot of widgets then the status bar becomes bloated and goes against the whole purpose of bloatfree stand-alone Window Manager setup. I don't know, maybe I'm just crazy. 

Either way once you close wldash like a typical launcher, the process gets terminated and no CPU cycles wasted when it's not open. It's  a little limited, but it does show pulse volume, time, calendar and laptop battery status.

I figured since there's no clean, non hacky way to show workspace indicators in niri, wldash serves me quite well and I got rid of waybar.

I hope it's useful to someone else

**Comment by lpnh** [comment]
Wow, I was impressed. Even after seeing the screenshot, I wasn't expecting it to *just work*. Zero configuration needed, with a very nice default. For someone who keeps the waybar hidden 99% of the time, this is more than enough. I already forked it, lol. Let's see how much further I can tweak it. Thanks for sharing this!
